### PR TITLE
added jwt auth

### PR DESCRIPTION
**Pull request by TarasYurkovskyi**

### CODE REVIEW

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private Optional<String> getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return Optional.of(bearerToken.substring(7));
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]: magic number

File: src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+
+        filterChain.doFilter(request, response);
+    }
+
+    private Optional<String> getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstore/security/JwtAuthenticationFilter.java]: Please make "Bearer " constant

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.example.bookstore.dto.user;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotEmpty;
+import lombok.Data;
+
+@Data
+public class UserLoginRequestDto {
+    @NotEmpty
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java]:
```suggestion
    @NotBlank
```

File: src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    @Email
+    private String email;
+    @NotEmpty
```
Comment (d1sam) [inline src/main/java/com/example/bookstore/dto/user/UserLoginRequestDto.java]:
```suggestion
    @NotBlank
```

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but one minor change should be done!

**Review by d1sam — APPROVED** [review]
Well done!

**Pull request merged**